docs/modules/ROOT/pages/usage.adoc

```diff
@@ -0,0 +1,297 @@
+    }
+}
+----
+
+=== Dependencies
+
+:access-component: xref:/api/access.adoc#AccessControlComponent[AccessControlComponent]
+:src5-component: xref:/api/introspection.adoc#SRC5Component[SRC5Component]
+
+Some components include dependencies of other components.
+Contracts that integrate components with dependencies must also include the component dependency.
+For instance, {access-component} depends on {src5-component}.
+Creating a contract with `AccessControlComponent` should look like this:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    use openzeppelin::access::accesscontrol::AccessControlComponent;
+    use openzeppelin::introspection::src5::SRC5Component;
+
+    component!(path: AccessControlComponent, storage: accesscontrol, event: AccessControlEvent);
+    component!(path: SRC5Component, storage: src5, event: SRC5Event);
+
+    // AccessControl
+    #[abi(embed_v0)]
+    impl AccessControlImpl =
+        AccessControlComponent::AccessControlImpl<ContractState>;
+    #[abi(embed_v0)]
+    impl AccessControlCamelImpl =
+        AccessControlComponent::AccessControlCamelImpl<ContractState>;
+    impl AccessControlInternalImpl = AccessControlComponent::InternalImpl<ContractState>;
+
+    // SRC5
+    #[abi(embed_v0)]
+    impl SRC5Impl = SRC5Component::SRC5Impl<ContractState>;
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        accesscontrol: AccessControlComponent::Storage,
+        #[substorage(v0)]
+        src5: SRC5Component::Storage
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
+        AccessControlEvent: AccessControlComponent::Event,
+        #[flat]
+        SRC5Event: SRC5Component::Event
+    }
+
+    (...)
+}
+----
+
+== Customization
+
+:security: xref:/usage.adoc#security[Security]
+
+WARNING: Customizing implementations and accessing component storage can potentially corrupt the state, bypass security checks, and undermine the component logic.
+*Exercise extreme caution*. See {security}.
+
+=== Custom implementations
+
+:erc20-component: xref:/api/erc20.adoc#ERC20Component[ERC20Component]
+:decimals: xref:/api/erc20.adoc#IERC20Metadata-decimals[decimals]
+:ierc20metadata: xref:/api/erc20.adoc#IERC20Metadata[IERC20Metadata]
+:custom-decimals: xref:/erc20.adoc#customizing_decimals[Customizing decimals]
+
+There may be instances where a contract requires different behavior from a component implementation.
+The contract can create a custom implementation by importing the appropriate interface and defining the new logic.
+
+In {erc20-component}, for example, the `ERC20MetadataImpl` provides a static {decimals} method which returns`18`.
+If a contract wanted to change the return value of `decimals`, the `impl` can no longer be embedded.
+Rather, the contract would have to create its own `impl` of the {ierc20metadata} interface and expose it in the contract.
+
+TIP: {custom-decimals} provides a guide for this exact scenario.
```

Comment:
i think we need to be way more explicit. this feedback came up twice today (from @ericglau while working in the Wizard, and by some users over telegram).

maybe we should expand this section showing how to make ERC20 transfers pausable